### ISSUE TITLE
[keyword-spacing] before: false with greater than

### ISSUE THREAD

**Issue by sam-s4s** [typescript-eslint/typescript-eslint#3535]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```js
{
  "rules": {
    'keyword-spacing': 'off',
    '@typescript-eslint/keyword-spacing': [
        'error',
        {
            "overrides": {
                "this": { "before": false },
                "new": { "before": false }
            }
        }
    ],
  }
}
```

```TS
if (this.refundAmount > this.maximumRefund) {
```

**Expected Result**
No error

**Actual Result**
283:41  error  Unexpected space(s) before "this"  @typescript-eslint/keyword-spacing

**Additional Info**
I wasn't sure what to expect from this configuration of the rule. I had to set "this" and "new" to before: false, because I don't want a space between casting angle brackets `<Thing>this.blah` or `<Thing>new Blah()`.
And I was happy when it seemed to work in almost all cases. Obviously I'm using `this.blah` in thousands of places, very commonly with spaces in front, and almost all of them work. The only exception is the "greater than" symbol, used for comparisons. Less than (and everything else) seems to work fine.

```
if (this.refundAmount < this.maximumRefund) { // the same thing with a less than works fine
```

So given it only gives an error with the exact combination `> this` I feel like it's a bug.

Would be similarly happy with the option to just completely disable for this in the overrides, possibly...

```js
{
  "rules": {
    'keyword-spacing': 'off',
    '@typescript-eslint/keyword-spacing': [
        'error',
        {
            "overrides": {
                "this": { "before": null },
                "new": { "before": null }
            }
        }
    ],
  }
}
```
(or a string "always" / "never" / "ignore"  instead of true / false)

*EDIT
I've also realised that this could be a problem with the base eslint rule, not typescript-eslint - if this is the case, I'm happy to recreate this issue there.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.23.0` |
| `@typescript-eslint/parser`        | `4.23.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `14.17.0` |

**Comment by bradzacher** [comment]
> I've also realised that this could be a problem with the base eslint rule, not typescript-eslint - if this is the case, I'm happy to recreate this issue there.

Correct - this is working as expected - as it works this way in the base ESLint rule.

https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGtleXdvcmQtc3BhY2luZzogWydlcnJvcicsIHtcbiAgICAgICAgICAgIFwib3ZlcnJpZGVzXCI6IHtcbiAgICAgICAgICAgICAgICBcInRoaXNcIjogeyBcImJlZm9yZVwiOiBmYWxzZSB9LFxuICAgICAgICAgICAgICAgIFwibmV3XCI6IHsgXCJiZWZvcmVcIjogZmFsc2UgfVxuICAgICAgICAgICAgfVxuICAgICAgICB9XSAqL1xuXG5pZiAodGhpcy5yZWZ1bmRBbW91bnQgPiB0aGlzLm1heGltdW1SZWZ1bmQpIHt9Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==

Please file an issue with them!

**Comment by sam-s4s** [comment]
Thank you, I will do this now :)
